File: codetemplate/tpl.py
#Implement your "nextMove" and "isValid" functions.
#You are free to add more functions or remove any function.
#If you are not clear about the rules, click the "Game Rules" button on the player bar.

def is_valid(board, move):
    # Check If the Move is Valid
    # Implement Your Checking Function
    return True
    
def next_move(board):
    # Implement Your AI Here
    if is_valid(board, 'UP'):
        return 'UP'
    
while True:
    # Read Game Board
    board = [[int(x) for x in raw_input().split()]for i in range(4)]
    # Compute Your Next Move
    move = next_move(board)
    # Print Your Next Move to the Standard Output
    print(move)

File: codetemplate/tpl.js
'use strict'

/*
 * Implement your "nextMove" and "isValid" functions.
 * You are free to add more functions or remove any function.
 * If you are not clear about the rules, click the "Game Rules" button on the player bar.
 */

const compute = (board) => {
	// Implement Your AI Here
	return "UP";
}

process.stdin.on('data', data => {
	// parse stdin to lines
	const lines = data.toString().split('\n');

	// game board
	const board = lines.slice(0, 4).map(l => l.split(' ').map(s => +s));

	// compute my move and print it to stdout
	console.log(compute(board));
});
